source/Locales/ContinuousMap/Homeomorphism-Properties.lagda

```diff
@@ -0,0 +1,85 @@
+--------------------------------------------------------------------------------
+title:          Properties of locale homeomorphisms
+author:         Ayberk Tosun
+date-started:   2024-04-18
+--------------------------------------------------------------------------------
+
+Some properties of locale homeomorphisms.
+
+\begin{code}[hide]
+
+{-# OPTIONS --safe --without-K #-}
+
+open import MLTT.Spartan hiding (𝟚; ₀; ₁)
+open import UF.Base
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Size
+open import UF.Subsingletons
+open import UF.UA-FunExt
+open import UF.Univalence
+
+module Locales.ContinuousMap.Homeomorphism-Properties
+        (ua : Univalence)
+        (pt : propositional-truncations-exist)
+        (sr : Set-Replacement pt)
+       where
+
+private
+ fe : Fun-Ext
+ fe {𝓤} {𝓥} = univalence-gives-funext' 𝓤 𝓥 (ua 𝓤) (ua (𝓤 ⊔ 𝓥))
+
+ pe : Prop-Ext
+ pe {𝓤} = univalence-gives-propext (ua 𝓤)
+
+open import Locales.ContinuousMap.Definition pt fe
+open import Locales.ContinuousMap.FrameIsomorphism-Definition pt fe
+open import Locales.ContinuousMap.Homeomorphism-Definition pt fe
+open import Locales.Frame pt fe
+open import Locales.SIP.FrameSIP ua pt sr
+open import Slice.Family
+open import UF.Equiv
+open import UF.Hedberg
+open import UF.Logic
+open import UF.Sets
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.SubtypeClassifier
+
+open AllCombinators pt fe
+open Locale
+
+\end{code}
+
+Homeomorphic locales are equal.
+
+\begin{code}
+
+homeomorphic-locales-are-equal : (X Y : Locale (𝓤 ⁺) 𝓤 𝓤) → X ≅c≅ Y → X ＝ Y
```

Comment:
I would make the parameters of the locales as general as possible throughout this file.

Reply:
I’ll do that in a separate PR in the future as to avoid spending any more time on this. As mentioned in PR #262, I wrote this down quickly to be able to prove a specific lemma about large and locally small spectral locales.

I will, however, write this down as a TODO for the reader who might be confused as to why this is done only for large and locally small locales.

Reply:
Up to you — just think it's easier to generalize now than in the future.

Reply:
That is a good point, but I’m a bit worried that it will require generalizing universes in a bunch of other places which will result in the PR becoming huge. At the moment, this PR achieves a well-defined goal in a self-contained way. I think it’s best if we merge this now, and move on to the generalization afterwards.